sled-agent/health-monitor/src/handle.rs

```diff
@@ -0,0 +1,56 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::health_checks::poll_smf_services_in_maintenance;
+
+use illumos_utils::svcs::SvcsInMaintenanceResult;
+use nexus_sled_agent_shared::inventory::HealthMonitorInventory;
+use slog::Logger;
+use slog::info;
+use tokio::sync::watch;
+
+#[derive(Debug, Clone)]
+pub struct HealthMonitorHandle {
+    // Return a String instead of a custom error type as inventory requires
+    // all types to be cloneable. The only error that could happen here is
+    // the failure to execute `svcs`, which is a `illumos_utils::ExecutionError`
+    // and this error cannot be cloned.
+    pub smf_services_in_maintenance_tx:
+        watch::Sender<Result<SvcsInMaintenanceResult, String>>,
```

Comment:
This handle probably only needs the receiver, right? I think that would clean up some of the clones inside `spawn()`, too.